ci-operator/step-registry/cucushift/installer/check/gcp/private/cucushift-installer-check-gcp-private-commands.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+GCP_REGION="${LEASED_RESOURCE}"
+
+GCP_BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ret=0
+
+echo "INFO: (1/4) Checking publish policy..."
+publish_policy=$(oc get configmap -n kube-system cluster-config-v1 -oyaml | grep 'publish:')
+echo "INFO: ${publish_policy}"
+if [[ "${publish_policy}" =~ "publish: Internal" ]]; then
+    echo "INFO: Publish policy check passed."
+else
+    echo "ERROR: Publish policy check failed."
+    ret=$(( $ret | 1 ))
+fi
+
+echo "INFO: (2/4) Checking if any dns record-sets in the base domain (public zone)..."
+base_domain_zone_name=$(gcloud dns managed-zones list --filter="dnsName=${GCP_BASE_DOMAIN}." --format="table(name)" | grep -v NAME)
+echo "INFO: base domain zone name '${base_domain_zone_name}'"
+gcloud dns record-sets list --zone "${base_domain_zone_name}" | grep "${CLUSTER_NAME}" && ret=$(( $ret | 2 ))
```

Comment:
> 1. filter results on server side via `--filter`, which can reduce network traffic.

FYI Generally it takes several seconds only and the time diff doesn't matter much I think. Anyway, in below outputs, it looks like "--filter" is a bit slower than "grep". 
```
[jiwei@jiwei ~]$ time gcloud dns record-sets list --zone qe | grep ci-op-0cglv5kc-e9af6
api.ci-op-0cglv5kc-e9af6.qe.gcp.devcluster.openshift.com.              A     60     34.172.128.253
*.apps.ci-op-0cglv5kc-e9af6.qe.gcp.devcluster.openshift.com.           A     30     35.192.169.65

real	0m2.913s
user	0m0.884s
sys	0m0.081s
[jiwei@jiwei ~]$ time gcloud dns record-sets list --zone qe --filter="name~ci-op-0cglv5kc-e9af6"
NAME                                                          TYPE  TTL  DATA
api.ci-op-0cglv5kc-e9af6.qe.gcp.devcluster.openshift.com.     A     60   34.172.128.253
*.apps.ci-op-0cglv5kc-e9af6.qe.gcp.devcluster.openshift.com.  A     30   35.192.169.65

real	0m3.580s
user	0m0.787s
sys	0m0.067s
[jiwei@jiwei ~]$ time gcloud dns record-sets list --zone qe | grep ci-op-0cglv5kc-e9af6
api.ci-op-0cglv5kc-e9af6.qe.gcp.devcluster.openshift.com.              A     60     34.172.128.253
*.apps.ci-op-0cglv5kc-e9af6.qe.gcp.devcluster.openshift.com.           A     30     35.192.169.65

real	0m2.592s
user	0m0.834s
sys	0m0.079s
[jiwei@jiwei ~]$ time gcloud dns record-sets list --zone qe --filter="name~ci-op-0cglv5kc-e9af6"
NAME                                                          TYPE  TTL  DATA
api.ci-op-0cglv5kc-e9af6.qe.gcp.devcluster.openshift.com.     A     60   34.172.128.253
*.apps.ci-op-0cglv5kc-e9af6.qe.gcp.devcluster.openshift.com.  A     30   35.192.169.65

real	0m4.906s
user	0m0.807s
sys	0m0.078s
[jiwei@jiwei ~]$

Reply:
> 2\. check the result content instead of return code, the benefit is more straightforward than `${ret} -ge N`

The current solution checks the grep exit code (zeor or non-zero), and meanwhile prints out the grep results automatically. Also include the grep exit code check into the final exit code of the script. 

And your suggestion would need to print the command output explicitly, and check literally the command output. 

That being said, why do you think your suggestion is more straightforward? Thanks! 

cc @jianlinliu

Reply:
> FYI Generally it takes several seconds only and the time diff doesn't matter much I think. Anyway, in below outputs, it looks like "--filter" is a bit slower than "grep".

Thanks for the AB tests, good to know.

> why do you think your suggestion is more straightforward? Thanks!

Personally, I think the readability of `if ${record_sets}-is-expected; then` is better than `if [ ${ret} -ge 2|4|8 ]; then`, especially we need to handle non-zero return value while enabling `errexit` option.

Reply:
Thank you, Yunfei! 

@jianlinliu What's your opinion? If you agree with Yunfei, I will update accordingly. Thanks!

Reply:
If we want more readability, I even prefer `$ret + 1` but not `$ret | {2|4|8}`, especially we also print out error message when some check failed. But I would like to leave this the case author to decide it, personally I think it is not a big deal.

Reply:
Okay, I prefer the current solution, thanks! 

_FYI_
- I didn't use `$ret + 1` because I want all checks getting executed and detect the success/failure of each.
- Comparing to `if ${record_sets}-is-expected; then`, it is easier to check the command exit code being zero or non-zero.